avocado/core/loader.py

```diff
@@ -250,19 +239,29 @@ def handle_exception(plugin, details):
                     try:
                         _test = loader_plugin.discover(reference, which_tests)
                         if _test:
+                            for item in _test:
+                                item[1]['loader'] = loader_plugin.name.upper()
                             tests.extend(_test)
                             handled = True
                             if not which_tests:
                                 break  # Don't process other plugins
+                        else:
+                            if which_tests:
+                                tests.append((test.MissingTest,
+                                             {'loader': loader_plugin.name.upper(),
+                                              'name': reference}))
                     except Exception as details:
-                        handle_exception(loader_plugin, details)
+                        if which_tests:
+                            tests.append((test.MissingTest,
```

Comment:
I think this is really worth fixing, so I created a  card to track it:

https://trello.com/c/TinXwGOB/996-remove-test-types-that-are-not-really-test-types